Src/RE2.Managed/Regex2.cs

```diff
@@ -255,7 +257,15 @@ public static unsafe bool Matches(string pattern, string text, out List<Dictiona
                         int submatchTextStartIndex = submatchRe2.Index;
                         int submatchLength = submatchRe2.Length;
 
-                        string submatchString = Encoding.UTF8.GetString(textUtf8Bytes, submatchTextStartIndex, submatchLength);
+                        string submatchString;
+                        if ((submatchTextStartIndex == -1) && (submatchLength == -1))
```

Comment:
>submatchTextStartIndex [](http://example.com/codeflow?start=29&length=22)

NIT: i would tend to initialize this local with string.empty and then use a single conditional to reassign the GetString call.

Small fix that reduces vertical lines in code without reallky impacting perf.

Reply:
See this PR:
[https://github.com/microsoft/sarif-pattern-matcher/pull/418](https://github.com/microsoft/sarif-pattern-matcher/pull/418)